File: mywebsite-parent/mywebsite-manager-web/src/main/webapp/angularjs/js/service/userService.js
app.service('userService', function ($http) {


    /**
     * 查询所有用户
     * @returns {*}
     */
    this.findAll = function () {
        return $http.get('../user/findAll.do');
    };

    /**
     * 条件分页搜索
     * @param searchEntity
     * @param page
     * @param rows
     * @returns {*}
     */
    this.search = function (searchEntity, page, rows) {
        return $http.post('../user/search.do?page=' + page + '&rows=' + rows, searchEntity);
    };

    /**
     * 根据id查询用户
     * @param id
     * @returns {*}
     */
    this.findOneById = function (id) {
        return $http.get('../user/findOneById.do?id=' + id);
    };

    /**
     * 更新用户
     * @param entity
     * @returns {*}
     */
    this.update = function (entity) {
        return $http.post('../user/update.do', entity);
    };


    /**
     * 添加用户
     * @param entity
     * @returns {*}
     */
    this.add = function (entity) {
        return $http.post('../user/add.do', entity);
    };

    /**
     * 批量删除
     * @param ids
     * @returns {*}
     */
    this.deleteIds = function (ids) {
        return $http.get('../user/deleteIds.do?ids='+ids);
    };
});

File: mywebsite-parent-v2.0/mywebsite-manager-web/src/main/java/xyz/lirui123/mywebsite/manager/service/UserService.java
package xyz.lirui123.mywebsite.manager.service;

import xyz.lirui123.mywebsite.pojo.TbUser;
import xyz.lirui123.mywebsite.response.ResponseResult;

import java.util.List;

public interface UserService  {

    /**
     * 查询所有
     * @return
     */
    public abstract List<TbUser> findAll();
    /**
     * 根据Id查询
     * @param id
     * @return
     */
    public abstract TbUser findOneById(Long id);

    /**
     * 根据邮箱查询
     * @param email
     * @return
     */
    public abstract TbUser findOneByEmail(String email);

    /**
     * 条件分页搜索
     * @param user
     * @param pageNum
     * @param pageSize
     * @return
     */
    public abstract ResponseResult search(TbUser user, int pageNum, int pageSize);

    /**
     * 添加用户
     * @param user
     * @return
     */
    public abstract ResponseResult add(TbUser user);

    /**
     * 修改用户
     * @param user
     * @return
     */
    public abstract ResponseResult update(TbUser user);

    /**
     * 删除用户
     * @param ids
     * @return
     */
    public abstract ResponseResult deleteIds(Long[] ids);
}
